apps/frontend/src/app/5_pages/BitocracyPage/components/NewProposalForm/components/TreasuryStep/components/Parameter/Parameter.tsx

```diff
@@ -0,0 +1,245 @@
+      <FormGroup
+        label={
+          <Paragraph size={ParagraphSize.base} className="font-medium">
+            {t(pageTranslations.treasuryAccount)}
+          </Paragraph>
+        }
+        labelElement="div"
+        className="w-full"
+        dataAttribute="proposal-treasury-account"
+      >
+        <div className="flex justify-between items-center">
+          <Select
+            value={parameter.target}
+            onChange={value => onChangeProperty('target', value)}
+            options={TREASURY_OPTIONS}
+            className="w-full"
+            dataAttribute="proposal-treasury-select"
+          />
+          {parameters.length > 1 && (
+            <Button
+              onClick={handleDeleteClick}
+              text={<Icon className="mx-4" icon={IconNames.X_MARK} />}
+              style={ButtonStyle.ghost}
+              className="text-sov-white ml-2 opacity-80 hover:opacity-100"
+            />
+          )}
+        </div>
+      </FormGroup>
+
+      <FormGroup
+        label={
+          <Paragraph size={ParagraphSize.base} className="font-medium mt-1">
+            {t(pageTranslations.recipientAddress)}
+          </Paragraph>
+        }
+        labelElement="div"
+        className="w-full"
+        dataAttribute="proposal-treasury-recipient-label"
+        errorLabel={errorAddressMessage}
+      >
+        <Input
+          value={parameter.parametersStepExtraData?.recipientAddress}
+          onChangeText={value =>
+            onChangeExtraProperty(TreasuryParameterType.recipientAddress, value)
+          }
+          dataAttribute="proposal-treasury-recipient-address"
+          classNameInput="min-h-10"
+          className="max-w-full"
+        />
+      </FormGroup>
+
+      <FormGroup
+        className="w-full mt-1"
+        dataAttribute="proposal-treasury-amount"
+        errorLabel={errorAmountMessage}
+      >
+        <div className="w-full flex flex-row justify-end mb-1">
+          <MaxButton
+            onClick={() => onChangeProperty('value', userBalance.toString())}
+            value={maximumAmount}
+            token={
+              parameter.parametersStepExtraData?.token || SupportedTokens.rbtc
+            }
+            dataAttribute="proposal-treasury-max-button"
+            label={t(pageTranslations.accountBalance)}
+          />
+        </div>
+        <div className="w-full flex flex-row justify-between items-center">
+          <AmountInput
+            value={parameter.value}
+            onChangeText={value => onChangeProperty('value', value)}
+            min={0}
+            label={t(translations.common.amount)}
+            className="w-full flex-grow-0 flex-shrink"
+            placeholder="0"
+          />
+          <Select
+            value={
+              parameter.parametersStepExtraData?.token || SupportedTokens.rbtc
+            }
+            onChange={value =>
+              onChangeExtraProperty(TreasuryParameterType.token, value)
+            }
```

Comment:
When token is changed, also update `signature`, for vaults if it's `rbtc` - `transferRbtc(address,uint256)`, if any other token - `transferTokens(address,address,uint256)`